src/tests/presets/test_erc721.cairo

```diff
@@ -0,0 +1,1165 @@
+
+    let short_ids = array![TOKEN_1, TOKEN_2].span();
+    let token_uris = array![URI_1, URI_2, URI_3].span();
+    state._mint_assets(OWNER(), short_ids, token_uris);
+}
+
+//
+// constructor
+//
+
+#[test]
+#[available_gas(2000000000)]
+fn test_constructor() {
+    let dispatcher = setup_dispatcher_with_event();
+
+    // Check interface registration
+    let mut interface_ids = array![ISRC5_ID, IERC721_ID, IERC721_METADATA_ID];
+    loop {
+        let id = interface_ids.pop_front().unwrap();
+        if interface_ids.len() == 0 {
+            break;
+        }
+        assert(dispatcher.supports_interface(id), 'Should support interface');
+    };
+
+    // Check token balance and owner
+    let mut tokens = array![TOKEN_1, TOKEN_2, TOKEN_3];
+    assert(dispatcher.balance_of(OWNER()) == TOKENS_LEN, 'Should equal TOKENS_LEN');
+    loop {
+        let token = tokens.pop_front().unwrap();
+        if tokens.len() == 0 {
+            break;
+        }
+        assert(dispatcher.owner_of(token) == OWNER(), 'Should be owned by OWNER');
+    };
+}
+
+#[test]
+#[available_gas(2000000000)]
+fn test_constructor_events() {
+    let dispatcher = setup_dispatcher_with_event();
+    let mut tokens = array![TOKEN_1, TOKEN_2, TOKEN_3];
+
+    loop {
+        let token = tokens.pop_front().unwrap();
+        if tokens.len() == 0 {
+            // Includes event queue check
+            assert_only_event_transfer(dispatcher.contract_address, ZERO(), OWNER(), token);
+            break;
+        }
+        assert_event_transfer(dispatcher.contract_address, ZERO(), OWNER(), token);
+    };
+}
+
+//
+// Getters
+//
+
+#[test]
+#[available_gas(20000000)]
+fn test_balance_of() {
+    let dispatcher = setup_dispatcher();
+    assert(dispatcher.balance_of(OWNER()) == TOKENS_LEN, 'Should return balance');
+}
+
+#[test]
+#[available_gas(20000000)]
+#[should_panic(expected: ('ERC721: invalid account', 'ENTRYPOINT_FAILED'))]
+fn test_balance_of_zero() {
+    let dispatcher = setup_dispatcher();
+    dispatcher.balance_of(ZERO());
+}
+
+#[test]
+#[available_gas(20000000)]
+fn test_owner_of() {
+    let dispatcher = setup_dispatcher();
+    assert(dispatcher.owner_of(TOKEN_1) == OWNER(), 'Should return owner');
+}
+
+#[test]
+#[available_gas(20000000)]
+#[should_panic(expected: ('ERC721: invalid token ID', 'ENTRYPOINT_FAILED'))]
+fn test_owner_of_non_minted() {
+    let dispatcher = setup_dispatcher();
+    dispatcher.owner_of(7);
+}
+
+#[test]
+#[available_gas(20000000)]
+#[should_panic(expected: ('ERC721: invalid token ID', 'ENTRYPOINT_FAILED'))]
+fn test_token_uri_non_minted() {
+    let dispatcher = setup_dispatcher();
+    dispatcher.token_uri(7);
+}
+
+#[test]
+#[available_gas(20000000)]
+fn test_get_approved() {
+    let dispatcher = setup_dispatcher();
+    let spender = SPENDER();
+    let token_id = TOKEN_1;
+
+    assert(dispatcher.get_approved(token_id) == ZERO(), 'Should return non-approval');
+
+    testing::set_contract_address(OWNER());
```

Comment:
I go back and forth with this. I like how it's explicit, but it's probably not terrible to assume the contract (aka caller) is `owner` unless specified otherwise. Will update